presto-native-execution/presto_cpp/main/PrestoExchangeSource.cpp

```diff
@@ -357,7 +366,9 @@ void PrestoExchangeSource::processDataResponse(
       queue_->enqueueLocked(nullptr, queuePromises);
     }
 
-    sequence_ = ackSequence;
+    if (ackSequenceOpt.has_value()) {
+      sequence_ = ackSequenceOpt.value();
+    }
```

Comment:
Have an else block and perform some finish checks?

Reply:
We also see empty sequence in case of get data size. Get data size and get data share the same response handler.